AnkiDroid/src/main/java/com/ichi2/utils/TypedFilter.kt

```
@@ -51,7 +52,13 @@ abstract class TypedFilter<T>(private val getCurrentItems: (() -> List<T>)) : Fi
     override fun publishResults(constraint: CharSequence?, results: FilterResults?) {
         // this is only ever called from performFiltering so we can guarantee the value is non-null
         // and can be cast to List<T>
-        val list = results!!.values as List<T>
+        val list = try {
+            results!!.values as List<T>
+        } catch (npe: NullPointerException) {
+            // allow publishResults to fail gracefully while reporting error
+            Timber.e(npe)
+            listOf()
+        }
```

I'd prefer to fail fast here. This should be caught further up the call stack where the context is understood.

---

Yea, I'm not sure exactly where else I could catch it without putting a generic catch in publishResults(). I'll just remove this completely on the assumption that the unit tests should stop this from happening